storage/objects/objects_test.go

```diff
@@ -479,14 +485,22 @@ func TestPostPolicyV4(t *testing.T) {
 		t.Fatalf("failed to compose HTTP request: %v", err)
 	}
 	req.Header.Set("Content-Type", mw.FormDataContentType())
+
+	// Dump the request for logging.
+	requestDump, err := httputil.DumpRequest(req, true)
+	if err != nil {
+		t.Logf("error dumping request for logging: %v", err)
+	}
+
 	res, err := http.DefaultClient.Do(req)
 	if err != nil {
 		t.Fatalf("client.Do: %v", err)
 	}
 	if g, w := res.StatusCode, 204; g != w {
 		blob, _ := httputil.DumpResponse(res, true)
-		t.Errorf("status code in response mismatch: got %d want %d\nBody: %s", g, w, blob)
+		t.Errorf("status code in response mismatch: got %d want %d\nBody: %s\nattempted request: %s", g, w, blob, string(requestDump))
```

Comment:
I would change this to put the request first, then the response, something like:

`...Request: %v\n\nResponse: %v\n", requestDump, blob)`